src/Integration.Vsix/CFamily/PortedFromJava/MsvcDriver.cs

```diff
@@ -256,10 +256,15 @@ public static Request ToRequest(CFamilyHelper.Capture[] captures)
                     }
                     else
                     {
-                        string file = Absolute(capture.Cwd, arg);
-                        if (file != null)
+                        // We know that the file to be analyzed is the last element.
+                        // If it is not the last element, it is the argument of an ignored option and it shouldn't be treated as a file
+                        if (args.Index == (capture.Cmd.Count - 1))
```

Comment:
I suggest a few changes:
1. Make the check after we leave the while loop (so then `args.Index == (capture.Cmd.Count - 1)` will always be true), instead of checking it for every iteration
2. Perhaps we should log the "else" and what option was it that fell through this if-else block.
3. There's no reason to use the list anymore. This code and the one below can be simplified to 

```
while (args.Index < capture.Cmd.Count) {..}

var file = Absolute(capture.Cwd, capture.Cmd[args.Index-1]);

if (file == null)
{
    throw new InvalidOperationException("No files to analyze");
}

request.File = file;

if (cpp == null)
{
    cpp = request.File.EndsWith(".c");
}
```

Reply:
1. The last branch will very rarely be hit (Only in case of this bug). the one that will be hit the most is:
```
                    else if (arg.StartsWith("/"))
                    {
                        // skip
                        args.Index++;

                    }
```
While putting the check outside the while will be cleaner, it will go against the functional logic that we are extracting all arguments from this switch. In fact, most of the branches can and should be extracted at some point that why I created [this](https://github.com/SonarSource/sonarlint-visualstudio/issues/1783).

2- No as we don't log the option that we ignore. We ignore them because they don't impact the analysis. I agree that we should log them if they aren't supported. like we did for `/ZW`

3- Yes, that makes sense